src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -32,17 +44,30 @@ public class EditCommandParser implements Parser<EditCommand> {
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, FLAG_NAME, FLAG_PHONE, FLAG_EMAIL, FLAG_ADDRESS, FLAG_TAG);
+                ArgumentTokenizer.tokenize(args, FLAG_NAME, FLAG_PHONE, FLAG_EMAIL, FLAG_ADDRESS, FLAG_TAG,
+                        FLAG_URL);
 
         Index index;
+        Id targetId;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            String preambleStr = argMultimap.getPreamble();
+            if (preambleStr.matches("^[A-Za-z].*")) {
```

Comment:
This would require a change in definition of `Id.VALIDATION_REGEX`, as well as ensuring `UUID.randomUUID()` also generates an ID with a leading letter